src/main/kotlin/com/hypto/iam/server/service/UserAuthService.kt

```diff
@@ -1,19 +1,62 @@
 package com.hypto.iam.server.service
 
+import com.hypto.iam.server.authProviders.AuthProviderRegistry
 import com.hypto.iam.server.db.repositories.OrganizationRepo
 import com.hypto.iam.server.db.repositories.UserAuthRepo
+import com.hypto.iam.server.db.repositories.UserRepo
 import com.hypto.iam.server.exceptions.EntityNotFoundException
+import com.hypto.iam.server.models.BaseSuccessResponse
 import com.hypto.iam.server.models.UserAuthMethod
 import com.hypto.iam.server.models.UserAuthMethodsResponse
+import com.hypto.iam.server.security.AuthMetadata
+import com.hypto.iam.server.security.AuthenticationException
+import com.hypto.iam.server.security.TokenCredential
+import com.hypto.iam.server.security.TokenType
+import com.hypto.iam.server.security.UserPrincipal
 import com.hypto.iam.server.utils.IamResources
 import com.hypto.iam.server.utils.ResourceHrn
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 
 class UserAuthServiceImpl : KoinComponent, UserAuthService {
     private val organizationRepo: OrganizationRepo by inject()
+    private val userRepo: UserRepo by inject()
     private val userAuthRepo: UserAuthRepo by inject()
 
+    override suspend fun createUserAuth(
```

Comment:
For a new linking request, will we get x-issuer header as 3rd party issuer_id or IAM's issuer_id?
If 3rd party id, then x-issuer header and Authorization header will belong to different providers. Right?